PostFreeSurfer/PostFreesurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,418 @@
+
+opts_AddMandatory '--path' 'StudyFolder' 'path' "folder containing all timepoins and templates" "--path"
+opts_AddMandatory '--subject'   'Subject'   'subject ID' "Subject label. Note: this is distinct subject label. Use separate labels for template and timepoints"
+opts_AddMandatory '--longitudinal-template'  'Template'  'FS template ID' "Longitudinal template ID (same as Freesurfer long template ID)"
+opts_AddMandatory '--sessions' 'Timepoints_cross' 'FS timepoint ID(s)' "Freesurfer timepoint ID(s). For timepoint (session)\
+                    processing, specify current timepoint. For template processing, must specify all timepoints, @ separated.\
+                    Timepoint ID and Session are synonyms in HCP structural pipelines."
+opts_AddMandatory '--template_processing' 'TemplateProcessing' 'create template flag' "0 if TP processing; 1 if template processing (must be run after all TP's)"
+
+#the following options have the same meaning as in PreFreesurferPipeline.
+opts_AddMandatory '--t1template' 'T1wTemplate' 'file_path' "MNI T1w template"
+opts_AddMandatory '--t1templatebrain' 'T1wTemplateBrain' 'file_path' "Brain extracted MNI T1wTemplate"
+opts_AddMandatory '--t1template2mm' 'T1wTemplate2mm' 'file_path' "MNI 2mm T1wTemplate"
+opts_AddMandatory '--t2template' 'T2wTemplate' 'file_path' "MNI T2w template"
+opts_AddMandatory '--t2templatebrain' 'T2wTemplateBrain' 'file_path' "Brain extracted MNI T2wTemplate"
+opts_AddMandatory '--t2template2mm' 'T2wTemplate2mm' 'file_path' "MNI 2mm T2wTemplate"
+opts_AddMandatory '--templatemask' 'TemplateMask' 'file_path' "Brain mask MNI Template"
+opts_AddMandatory '--template2mmmask' 'Template2mmMask' 'file_path' "Brain mask MNI 2mm Template"
+opts_AddMandatory '--fnirtconfig' 'FNIRTConfig' 'file_path' "FNIRT 2mm T1w Configuration file"
+
+#needed to transform Freesurfer mask to MNI space.
+opts_AddMandatory '--freesurferlabels' 'FreeSurferLabels' 'file' "location of FreeSurferAllLut.txt"
+
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var FSLDIR
+log_Check_Env_Var HCPPIPEDIR_Global
+
+# Naming Conventions
+T1wImage="T1w"
+T1wImageBrainMask="brainmask_fs"
+T1wFolder="T1w" #Location of T1w images
+T2wImage="T2w"
+T2wFolder="T2w" #Location of T2w images
+MNI_hires_template="$HCPPIPEDIR/global/templates/MNI152_T1_0.8mm.nii.gz"
+
+IFS='@' read -r -a timepoints <<< "$Timepoints_cross"
+Timepoint_cross=${timepoints[0]}
+
+
+#########################################################################################
+# Organizing and cleaning up the folder structure
+#########################################################################################
+if (( TemplateProcessing == 0 )); then 
```

Comment:
I tend to spell this with a "not" rather than do an "equals zero", I think it is easier to parse.  There is a drawback in the syntax, though, in that the space after the `!` is absolutely required (or bash will treat it as a history expansion and it will do weird things).  I also try to name my booleans so it reads more like english.  This is more of a suggestion, you can stay with what you have if you want.
```suggestion
if ((! TemplateMode)); then
```

Reply:
Ok